rticonnextdds_connector/rticonnextdds_connector.py

```diff
@@ -279,14 +283,30 @@ def getString(self, index, fieldName):
 
 		return str_value
 
-	def getDictionary(self, index):
+	def getDictionary(self, index, member_name = None):
 		if type(index) is not int:
 			raise ValueError("index must be an integer")
 		if index < 0:
 			raise ValueError("index must be positive")
-		#Adding 1 to index because the C API was based on Lua where indexes start from 1
+		# Adding 1 to index because the C API was based on Lua where indexes start from 1
 		index = index + 1
-		jsonStrPtr = rtin_RTIDDSConnector_getJSONSample(self.input.connector.native,tocstring(self.input.name),index)
+		if member_name is None:
+			jsonStrPtr = rtin_RTIDDSConnector_getJSONSample(
+				self.input.connector.native,
+				tocstring(self.input.name),
+				index)
+		elif not isinstance(member_name, str):
+			raise ValueError("member_name must be a string")
+		else:
+			jsonStrPtr = ctypes.c_char_p()
+			retcode = rtin_RTIDDSConector_getJSONMember(
+				self.input.connector.native,
+				tocstring(self.input.name),
+				index,
+				tocstring(member_name),
+				ctypes.byref(jsonStrPtr))
+			_check_retcode(retcode)
```

Comment:
I realised that _check_retcode does not actually print the retcode if there is a failure - do you think it should?

Reply:
No, it should only throw an exception. If unhandled, py will print the exception message.